### PR TITLE
Restoring sharingLinkToRedeem property in main

### PR DESCRIPTION
**Pull request by sonalivdeshpande**
The PR aims to revert the changes to the deprecated property `sharingLinkToRedeem` in main. The property was removed from `IOdspResolvedUrl` interface in 0.52 release. Restoring the property both in main and 0.52 release.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 1]
```diff
@@ -104,7 +104,14 @@ export async function fetchSnapshotWithRedeem(
             }, error);
             await redeemSharingLink(odspResolvedUrl, storageTokenFetcher, logger);
             const odspResolvedUrlWithoutShareLink: IOdspResolvedUrl =
-            { ...odspResolvedUrl, shareLinkInfo: { ...odspResolvedUrl.shareLinkInfo, sharingLinkToRedeem: undefined } };
+            { ...odspResolvedUrl, sharingLinkToRedeem: undefined };
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:107]: nit: Tab space missing.

**Comment by jatgarg** [comment]
Could you remove usage of odspResolvedUrl.shareLinkInfo?.sharingLinkToRedeem to odspResolvedUrl.sharingLinkToRedeem in fetchsnapshot.ts file where it is. I am not able to comment at the exact place.

**Review by jatgarg — APPROVED** [review]

**Comment by jatgarg** [comment]
Looks good now. Thanks.

**Comment by sdeshpande3** [comment]
> Could you please remove usage of odspResolvedUrl.shareLinkInfo?.sharingLinkToRedeem to odspResolvedUrl.sharingLinkToRedeem in fetchsnapshot.ts file where it is. I am not able to comment at the exact place.

Do we want to check the property in the assert:
`assert(resolvedUrl.shareLinkInfo?.sharingLinkToRedeem !== undefined, "Sharing link should be set in resolved url");`

**Comment by jatgarg** [comment]
> > Could you please remove usage of odspResolvedUrl.shareLinkInfo?.sharingLinkToRedeem to odspResolvedUrl.sharingLinkToRedeem in fetchsnapshot.ts file where it is. I am not able to comment at the exact place.
> 
> Do we want to check the property in the assert: `assert(resolvedUrl.shareLinkInfo?.sharingLinkToRedeem !== undefined, "Sharing link should be set in resolved url");`

No just resolvedUrl.sharingLinkToRedeem is fine.

File: packages/drivers/odsp-driver/src/test/odspDriverUrlResolverForShareLink.spec.ts [hunk 2]
```diff
@@ -183,7 +183,6 @@ describe("Tests for OdspDriverUrlResolverForShareLink resolver", () => {
                 { url: url.toString(), headers: { [SharingLinkHeader.isSharingLinkToRedeem]: true } });
         });
         assert(resolvedUrl.sharingLinkToRedeem !== undefined, "Sharing link should be set in resolved url");
-        assert(resolvedUrl.shareLinkInfo?.sharingLinkToRedeem !== undefined, "Sharing link should be set in resolved url");
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/test/odspDriverUrlResolverForShareLink.spec.ts:185]: Ohh. this you can keep. It is in unit test.

**Review by agarwal-navin — APPROVED** [review]

**Pull request merged**